File: JavaScript/1160.js
/**
 * @no 1160
 * @name Find Words That Can Be Formed by Characters
 */
const countCharacters = function(words, chars) {
    const count = [];
    for (let i = 0; i < 26; i++) { count.push(0); }
    for (let i = 0; i < chars.length; i++) {
        count[chars.charCodeAt(i) - 'a'.charCodeAt(0)] += 1;
    }

    let result = 0;
    words.forEach(word => {
        const lCount = [];
        count.forEach(c => { lCount.push(c); });
        for (let i = 0; i < word.length; i++) {
            lCount[word.charCodeAt(i) - 'a'.charCodeAt(0)] -= 1;
            if (lCount[word.charCodeAt(i) - 'a'.charCodeAt(0)] < 0) { return; }
        }
        return result += word.length;
    });
    return result;
};

console.log(countCharacters(["cat","bt","hat","tree"], "atach"));

File: Java/1160.java
/**
 * @no 1160
 * @name Find Words That Can Be Formed by Characters
 */
class Solution {
    public int countCharacters(String[] words, String chars) {
        int[] hashMap = new int[26];
        for (char c : chars.toCharArray()) {
            hashMap[c - 'a']++;
        }

        int result = 0;
        NEXT_WORD:
        for (String word : words) {
            int[] temp = copyArray(hashMap);
            for (char c : word.toCharArray()) {
                if (temp[c - 'a'] <= 0) {
                    continue NEXT_WORD;
                } else {
                    temp[c - 'a']--;
                }
            }
            result += word.length();
        }
        return result;
    }

    private int[] copyArray(int[] array) {
        int[] result = new int[array.length];
        for (int i = 0; i < array.length; i++) {
            result[i] = array[i];
        }
        return result;
    }
}